modules/samples/firealarm/component/plugin/src/main/java/org/homeautomation/firealarm/plugin/impl/dao/impl/DeviceTypeDAOImpl.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.homeautomation.firealarm.plugin.impl.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.homeautomation.firealarm.plugin.constants.DeviceTypeConstants;
+import org.homeautomation.firealarm.plugin.exception.FirealarmPluginException;
+import org.homeautomation.firealarm.plugin.impl.dao.DeviceTypeDAO;
+import org.homeautomation.firealarm.plugin.impl.dao.util.DeviceTypeUtils;
+import org.wso2.carbon.device.mgt.common.Device;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Device Dao for Firealarm Devices.
+ */
+public class DeviceTypeDAOImpl {
```

Comment:
wrong wording for the class, if it's an impl, normally this should be extending some base class.